frontend/src/api/apiCalls.ts

```diff
@@ -1,28 +1,42 @@
+import axios from 'axios';
+import { log } from 'console';
+
 import { API_ENDPOINTS } from './apiConfig';
 
-export async function validateSearchResult() : Promise<boolean> {
+import { ItemRecommendationBoxData } from 'src/models/ItemRecommendationBoxData';
+
+
+interface ValidSearchPromptResponse {
+  status: string;
+}
+export async function validateSearchPrompt(prompt : string) : Promise<boolean> {
   try {
-    const response = await fetch(API_ENDPOINTS.validateSearchResult, {
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-    });
-    if(!response.ok) {
-      throw new Error('Failed to fetch results');
+    const response = await axios.post<ValidSearchPromptResponse>(API_ENDPOINTS.validateSearchResult, { prompt });
```

Comment:
async function validateSearchPrompt(prompt: string): Promise<boolean> {
  try {
    const response = await axios.post(API_ENDPOINTS.validateSearchResult, { prompt });
    return response.data.status === 'success';
  } catch (error) {
    console.error('Error validating search prompt:', error); // Log the entire error for debugging
  }
  return false;
}